torch/csrc/Device.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <Python.h>
+#include "torch/csrc/utils/device.h"
+
+struct THPDevice {
+  PyObject_HEAD
+  torch::DeviceType device_type;
+  int64_t device_index;
```

Comment:
nit: not a blocker for this PR, because our C code is fluid, but would be nice to drop the `device_` prefix in here as well.

Reply:
Also, any reason why this doesn't keep `torch::Device` as a field? That would ensure they will be in sync

Reply:
I thought it was nice to have them not in sync :).  Because e.g. you don't care about the AutoGPU index on the python side, but you do on the torch side and I wanted to make that clear.  But there's probably a better way to do it, I'll think about it.